pkg/storage/azure/azure.go

```diff
@@ -153,23 +153,32 @@ func (d *driver) accountExists(storageAccountsClient storage.AccountsClient, acc
 	)
 }
 
-func (d *driver) createStorageAccount(storageAccountsClient storage.AccountsClient, resourceGroupName, accountName, location string) error {
+func (d *driver) createStorageAccount(storageAccountsClient storage.AccountsClient, resourceGroupName, accountName, location, cloudName string) error {
 	klog.Infof("attempt to create azure storage account %s (resourceGroup=%q, location=%q)...", accountName, resourceGroupName, location)
 
+	kind := storage.StorageV2
+	params := &storage.AccountPropertiesCreateParameters{
+		AllowBlobPublicAccess: to.BoolPtr(false),
+		MinimumTLSVersion:     storage.TLS12,
+	}
+
+	if strings.EqualFold(cloudName, "AZURESTACKCLOUD") {
+		// It seems Azure Stack Hub does not support new API.
+		kind = storage.Storage
+		params = &storage.AccountPropertiesCreateParameters{}
+	}
```

Comment:
@patrickdillon does it make sense?

Reply:
Well, I'm not sure. It seems like ASH supports 2019-06-01, and therefore should support `AllowBlobPublicAccess` and  `MinimumTLSVersion`. (I don't know about `storage.StorageV2` vs `storage.Storage`. )

Are you basing the removal of these two fields on the fact that the hybrid profile does not have these fields or have you tested and get an error when those fields are included? I would expect that if this code works, you should be able to include those two fields with ASH.

Hopefully this comment is not too unclear.

Reply:
I tried and it didn't work. It returned

> storage.AccountsClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidParameter" Message="Parameters are not valid in the request" Target="" Details=[{"code":"AccountPropertyCannotBeUpdated","message":"Property storageAccount.properties.allowBlobPublicAccess that cannot be updated for the storage account was specified in the request.","target":"storageAccount.properties.allowBlobPublicAccess"},{"code":"AccountPropertyCannotBeUpdated","message":"Property storageAccount.properties.minimumTlsVersion that cannot be updated for the storage account was specified in the request.","target":"storageAccount.properties.minimumTlsVersion"}]

Reply:
Ok. I think this explains why the hybrid profile uses `2017-10-01`, which doesn't have these fields.

Yeah, with your explanation, this code makes sense to me.